server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterAllocationSimulationTests.java

```diff
@@ -0,0 +1,549 @@
+
+                final var totalShardCount = sizeMetric.totalShardCount;
+                final var meanShardCount = totalShardCount * 1.0 / sizeMetric.count;
+                final var maxShardCount = sizeMetric.maxShardCount;
+                final var overageShardCount = maxShardCount - meanShardCount;
+                final var overageShardCountRatio = new RatioValue(Math.ceil((1000.0 * overageShardCount) / meanShardCount) / 10.0);
+
+                final var meanSizeBytes = sizeMetric.totalSizeBytes / sizeMetric.count;
+                final var maxSizeBytes = sizeMetric.maxSizeBytes;
+                final var overageBytes = maxSizeBytes - meanSizeBytes;
+                final var overageBytesRatio = new RatioValue(Math.ceil((1000.0 * overageBytes) / meanSizeBytes) / 10.0);
+                final var meanWriteLoad = sizeMetric.totalWriteLoad / sizeMetric.count;
+                final var maxWriteLoad = sizeMetric.maxWriteLoad;
+                final var overageWriteLoad = maxWriteLoad - meanWriteLoad;
+                final var overageWriteLoadRatio = new RatioValue(Math.ceil((1000.0 * overageWriteLoad) / meanWriteLoad) / 10.0);
+                assert overageBytes >= 0;
+
+                results.startObject();
+
+                results.field("tier", tier);
+                results.field("nodes", nodeCountByTier.get(tier));
+
+                results.startObject("shard_count");
+                results.field("total", totalShardCount);
+                results.field("mean", meanShardCount);
+                results.field("max", maxShardCount);
+                results.field("overage", overageShardCount);
+                results.field("overage_percent", overageShardCountRatio.formatNoTrailingZerosPercent());
+                results.endObject(); // shard_count
+
+                results.startObject("data_per_node");
+                bytesField(results, "capacity", nodeSizeBytes);
+                bytesField(results, "mean", meanSizeBytes);
+                bytesField(results, "max", maxSizeBytes);
+                bytesField(results, "overage", overageBytes);
+                results.field("overage_percent", overageBytesRatio.formatNoTrailingZerosPercent());
+                results.endObject(); // data_per_node
+
+                results.startObject("write_load_per_node");
+                results.field("indexing_threads", indexingThreadsByTier.get(tier));
+                results.field("mean", meanWriteLoad);
+                results.field("max", maxWriteLoad);
+                results.field("overage", overageWriteLoad);
+                results.field("overage_percent", overageWriteLoadRatio.formatNoTrailingZerosPercent());
+                results.endObject(); // write_load_per_node
+
+                results.endObject();
+            }
+
+            results.endArray(); // tiers
+
+            results.endObject();
+            results.flush();
+            logger.info("\n\n{}\n\n", bos.bytes().utf8ToString());
```

Comment:
Do we want to log those results in all executions?

Reply:
Not really I guess - nobody will see them unless something fails but any failure likely happens before this point anyway. Quietened down the logging in f1156a544e8d03fc7987bae9e0882d6164980508.